app/controllers/orders_controller.rb

```diff
@@ -1,8 +1,14 @@
 class OrdersController < ApplicationController
 
   def index
-    @user = User.find(params[:user_id])
-    @orders = @user.orders
+
+    if @user = current_user
```

Comment:
Yeah, this is pretty much what I was thinking of, the only way it might be cleaned up is 

```
if current_user
  @orders = current_user.orders
else
  #flash message and redirect
end
```

Otherwise I think it's ready to merge